src/main/java/taxi/service/AuthenticationServiceImpl.java

```
@@ -0,0 +1,25 @@
+package taxi.service;
+
+import taxi.exception.AuthenticationException;
+import taxi.lib.Injector;
+import taxi.lib.Service;
+import taxi.model.Driver;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    private static final Injector injector = Injector.getInstance("taxi");
+    private final DriverService driverService
+            = (DriverService) injector.getInstance(DriverService.class);
+
+    @Override
+    public Driver login(String login, String password) throws AuthenticationException {
+        Driver driver = driverService.findByLogin(login);
+        if (driver == null) {
+            throw new AuthenticationException("Wrong login or password");
+        }
+        if (driver.getPassword().equals(password)) {
+            return driver;
+        }
+        throw new AuthenticationException("Wrong login or password");
```

rewrite your logic, so you need to throw the exception only one time